test/page-wait-for-response.spec.ts

```diff
@@ -56,6 +56,19 @@ it('should work with predicate', async ({page, server}) => {
   expect(response.url()).toBe(server.PREFIX + '/digits/2.png');
 });
 
+it('should work with async predicate', async ({page, server}) => {
+  await page.goto(server.EMPTY_PAGE);
+  const [response] = await Promise.all([
+    page.waitForResponse(async response => response.url() === server.PREFIX + '/digits/2.png'),
```

Comment:
Let's do a more realistic one? For example, load `one-style.html` and wait for response with `background-color` in the body.